src/gui/ocsjob.h

```diff
@@ -0,0 +1,122 @@
+/*
+ * Copyright (C) by Roeland Jago Douma <roeland@famdouma.nl>
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; version 2 of the License.
+ *
+ * This program is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY
+ * or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License
+ * for more details.
+ */
+
+#ifndef OCSJOB_H
+#define OCSJOB_H
+
+#include "accountfwd.h"
+#include "abstractnetworkjob.h"
+
+#include <QVector>
+#include <QList>
+#include <QPair>
+#include <QUrl>
+
+namespace OCC {
+
+/**
+ * @brief The OcsShareJob class
+ * @ingroup gui
+ *
+ * Base class for jobs that talk to the OCS endpoints on the server.
+ * All the comminication logic is handled in this class.
```

Comment:
s/comminication/communication/
I will just add low-value comments by finding the comment typos, and along the way maybe some day be able to learn how this beast all fits together and then contribute something actually useful :)